Widescreen patch
Old: Feyrbrand's model disappears after about a 90 degree rotation.
New: Feyrbrand continues to rotate and walk off screen
--- original
+++ modified
@@ -162,16 +162,18 @@
 call 97, 0x7
 mov 0x3c, stor[8]
 wait stor[8]
-call 99, 0x1
-mov 0x1f4, stor[24]
+call 97, 0
+mov 0x3e8, stor[24]
 mov 0xd2, stor[25]
-mov 0xc8, stor[26]
+mov 0xa8, stor[26]
 mov 0x5, stor[27]
 mov 0xffffffff, stor[28]
 gosub inl[:LABEL_2]
-mov 0x0, stor[24]
-mov 0x28, stor[25]
-gosub inl[:LABEL_7]
+mov 0x320, stor[26]
+mov 0xa8, stor[28]
+mov 0x50, stor[24]
+mov 0xffffffff, stor[20]
+gosub inl[:ROTATE]
 mov 0x3, stor[8]
 wait stor[8]
 call 2, 0x11, 0x1
@@ -183,8 +185,8 @@
 call 2, 0x11, 0x1
 mov 0x2, stor[8]
 wait stor[8]
+call 97, 1
 call 236, 0x8, 0xf, 0x0, 0x0, 0x0, 0x0
-call 106, 0x1
 mov 0x1c, stor[8]
 wait stor[8]
 call 2, 0x11, 0x1
@@ -223,6 +225,32 @@
 gosub inl[:LABEL_1]
 call 230, 0x46, 0x0
 rewind
+ROTATE:
+call 102, stor[0], stor[29], stor[30], stor[31]
+sub stor[29], stor[26]
+sub stor[31], stor[28]
+atan2_12 stor[26], stor[28], stor[27]
+call 104, stor[0], stor[29], stor[30], stor[31]
+and 0xfff, stor[30]
+sub stor[30], stor[27]
+add 0x800, stor[27]
+LT_N360_CHECK:
+jmp_cmp >=, stor[27], 0xfffff800, inl[:GT_P360_CHECK]
+add 0x1000, stor[27]
+jmp inl[:ROTATE_0]
+GT_P360_CHECK:
+jmp_cmp <, stor[27], 0x800, inl[:ROTATE_0]
+sub 0x1000, stor[27]
+ROTATE_0:
+call 120, stor[0], 0x0, stor[27], 0x0, stor[24]
+jmp_cmp ==, stor[20], 0xffffffff, inl[:RETURN]
+jmp_cmp ==, stor[20], 0xfffffffe, inl[:ROTATE_WAIT]
+call 99, 0
+call 97, stor[20]
+ROTATE_WAIT:
+wait stor[24]
+RETURN:
+return
 LABEL_17:
 call 97, 0x2
 mov 0x208, stor[24]
